### PR TITLE
Documentation for core opentracing package

### PR DESCRIPTION
**Pull request by benweitzman**

### CODE REVIEW

**Comment by kim** [comment]
Awesome, thank you so much! I’ll go over it hunting for typos and merge asap.

File: opentracing/OpenTracing/Reporting/Pure.hs [hunk 1]
```diff
@@ -17,21 +23,28 @@ import Data.IORef
...
     { siz :: Maybe Word32
     , vec :: IORef [FinishedSpan]
     }
 
+-- | Construct a new `memReporter` environment that can store an unbounded
+-- seequence of `FinishedSpan`s.
 newMem :: IO Mem
 newMem = Mem Nothing <$> newIORef []
 
+-- | Construct a new `memReprioerter` environment that stores a bounded
```
Comment (kim) [inline opentracing/OpenTracing/Reporting/Pure.hs]:
```suggestion
-- | Construct a new `memReporter` environment that stores a bounded
```

File: opentracing/OpenTracing/Span.hs [hunk 2]
```diff
@@ -1,3 +1,8 @@
+{-|
+Module: OpenTracing.Span
+
+Data types and functions for manipulation [spans](https://github.com/opentracing/specification/blob/master/specification.md#span)
```
Comment (kim) [inline opentracing/OpenTracing/Span.hs]:
```suggestion
Data types and functions for manipulating [spans](https://github.com/opentracing/specification/blob/master/specification.md#span)
```

File: opentracing/OpenTracing/Span.hs [hunk 3]
```diff
@@ -85,13 +90,22 @@ import OpenTracing.Tags
 import OpenTracing.Types
 import Prelude                hiding (span)
 
-
+-- | A `SpanContext` is the data that uniquely identifies a span
+-- and context in which it occurs. Spans occur in traces, which form
```
Comment (kim) [inline opentracing/OpenTracing/Span.hs]:
```suggestion
-- and the context in which it occurs. Spans occur in traces, which form
```

File: opentracing/OpenTracing/Reporting/Batch.hs [hunk 4]
```diff
@@ -56,25 +71,34 @@ batchOptions f = BatchOptions
...
 
 makeLenses ''BatchOptions
 
 
+-- | The environment of a batch reporter.
 data BatchEnv = BatchEnv
     { envQ   :: TQueue FinishedSpan
+    -- ^ The queue of spans to be reported
     , envRep :: Async ()
+    -- ^ Asyncronous consumer of the queue
```
Comment (kim) [inline opentracing/OpenTracing/Reporting/Batch.hs]:
```suggestion
    -- ^ Asynchronous consumer of the queue
```

File: opentracing/OpenTracing/Span.hs [hunk 5]
```diff
@@ -85,13 +90,22 @@ import OpenTracing.Tags
...
     { ctxTraceID      :: TraceID
+    -- ^ A trace identifier. Trace ids are globally unique
     , ctxSpanID       :: Word64
+    -- ^ A span identifier. Span identifiers are unique to their trace.
     , ctxParentSpanID :: Maybe Word64
+    -- ^ Spans without a parent are known as "root spans"
     , _ctxSampled     :: Sampled
+    -- ^ Whether or not this span is going to be reported.
     , _ctxBaggage     :: HashMap Text Text
+    -- ^ Baggage is arbitrary key:value pars that cross process boundaries.
```
Comment (kim) [inline opentracing/OpenTracing/Span.hs]:
```suggestion
    -- ^ Baggage is arbitrary key:value pairs that cross process boundaries.
```

File: opentracing/OpenTracing/Span.hs [hunk 6]
```diff
@@ -108,11 +122,21 @@ instance ToJSON SpanContext where
...
         ]
 
+-- | A wrapper for a value that was produced by a traced computation.
+--
+-- @since 0.1.0.0
 data Traced a = Traced
     { tracedResult :: a
+    -- ^ The raw value produced
     , tracedSpan   :: ~FinishedSpan
+    -- ^ The resulting span that was creating
```
Comment (kim) [inline opentracing/OpenTracing/Span.hs]:
```suggestion
    -- ^ The resulting span that was created
```

File: opentracing/OpenTracing/Span.hs [hunk 7]
```diff
@@ -125,6 +149,17 @@ _IsSampled = iso (bool NotSampled Sampled) $ \case
     Sampled    -> True
     NotSampled -> False
 
+-- | A reference from one span to another. Spans can be related in two ways:
+--
+--   * `ChildOf` indicates that the parent span is dependent on the child span in order
+--      to produce its own result.
+--
+--   * `FollowersFrom` indicates that there is no dependence relation, perhaps the
```
Comment (kim) [inline opentracing/OpenTracing/Span.hs]:
```suggestion
--   * `FollowsFrom` indicates that there is no dependence relation, perhaps the
```

File: opentracing/OpenTracing/Reporting/Pure.hs (hunk 1 above)
Comment (benweitzman) [reply opentracing/OpenTracing/Reporting/Pure.hs]: Wow this was quite the typo

**Pull request closed without merging**